tests/testthat/test-chop.R

```diff
@@ -309,6 +309,33 @@ test_that("unchopping drops outer names", {
   expect_named(out$col, NULL)
 })
 
+test_that("unchop works on foreign list types recognized by `vec_is_list()` (#1327)", {
```

Comment:
The 4 test files here have fairly similar tests, but it felt important to test them all separately because each one does different things before it finally gets down to either `df_unchop()` or `col_simplify()` and I wanted to make sure that nothing else along the way broke support for these foreign lists.